x/ccv/provider/keeper/relay.go

```diff
@@ -213,57 +216,68 @@ func (k Keeper) EndBlockCIS(ctx sdk.Context) {
+		return channeltypes.NewErrorAcknowledgement(err.Error())
 	}
 
-	ack := channeltypes.NewResultAcknowledgement([]byte{byte(1)})
-	return ack
+	// apply slashing
+	k.HandleSlashPacket(ctx, chainID, data)
+
+	return channeltypes.NewResultAcknowledgement([]byte{byte(1)})
 }
 
-// HandleSlashPacket slash and jail a misbehaving validator according the infraction type
-func (k Keeper) HandleSlashPacket(ctx sdk.Context, chainID string, data ccv.SlashPacketData) (success bool, err error) {
-	// map VSC ID to infraction height for the given chain ID
-	var infractionHeight uint64
-	var found bool
-	if data.ValsetUpdateId == 0 {
-		infractionHeight, found = k.GetInitChainHeight(ctx, chainID)
-	} else {
-		infractionHeight, found = k.GetValsetUpdateBlockHeight(ctx, data.ValsetUpdateId)
-	}
+// ValidateSlashPacket validates a recv slash packet before it is
+// handled or persisted in store. An error is returned if the packet is invalid,
+// and an error ack should be relayed to the sender.
+func (k Keeper) ValidateSlashPacket(ctx sdk.Context, chainID string,
+	packet channeltypes.Packet, data ccv.SlashPacketData) error {
 
+	_, found := k.getMappedInfractionHeight(ctx, chainID, data.ValsetUpdateId)
 	// return error if we cannot find infraction height matching the validator update id
 	if !found {
-		return false, fmt.Errorf("cannot find infraction height matching the validator update id %d for chain %s", data.ValsetUpdateId, chainID)
+		return fmt.Errorf("cannot find infraction height matching "+
+			"the validator update id %d for chain %s", data.ValsetUpdateId, chainID)
+	}
+
+	if data.Infraction != stakingtypes.DoubleSign && data.Infraction != stakingtypes.Downtime {
+		return fmt.Errorf("invalid infraction type: %s", data.Infraction)
 	}
 
-	// get the validator
+	return nil
+}
+
+// HandleSlashPacket potentially slashes, jails and/or tombstones
+// a misbehaving validator according to infraction type.
+func (k Keeper) HandleSlashPacket(ctx sdk.Context, chainID string, data ccv.SlashPacketData) {
+
+	// Get the validator
 	consAddr := sdk.ConsAddress(data.Validator.Address)
+	// TODO: Key assignment will change the following line
 	validator, found := k.stakingKeeper.GetValidatorByConsAddr(ctx, consAddr)
 
 	// make sure the validator is not yet unbonded;
 	// stakingKeeper.Slash() panics otherwise
 	if !found || validator.IsUnbonded() {
-		// TODO add warning log message
-		// fmt.Sprintf("consumer chain %s trying to slash unbonded validator %s", chainID, consAddr.String())
-		return false, nil
+		// if validator is not found or is unbonded drop slash packet and log error
+		k.Logger(ctx).Error("validator not found or is unbonded", "validator", data.Validator.Address)
+		return
 	}
 
-	// tombstoned validators should not be slashed multiple times
+	// tombstoned validators should not be slashed multiple times.
 	if k.slashingKeeper.IsTombstoned(ctx, consAddr) {
-		return false, nil
+		// Drop packet if validator is tombstoned.
```

Comment:
Maybe log an info message here. See https://github.com/cosmos/cosmos-sdk/blob/47f46643affd7ec7978329c42bac47275ac7e1cc/x/slashing/keeper/infractions.go#L117 and https://github.com/cosmos/cosmos-sdk/blob/47f46643affd7ec7978329c42bac47275ac7e1cc/x/evidence/keeper/infraction.go#L79.